library/core/src/num/int_macros.rs

```diff
@@ -2000,5 +2000,176 @@ macro_rules! int_impl {
         pub const fn max_value() -> Self {
             Self::MAX
         }
+
+        /// Adds to the current value, returning the previous value.
+        ///
+        /// This operation wraps around on overflow.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// #![feature(int_fetch_ops)]
+        #[doc = concat!("let mut foo: ", stringify!($SelfT), " = 0;")]
+        /// assert_eq!(foo.fetch_add(10), 0);
+        /// assert_eq!(foo, 10);
+        /// ```
+        #[inline]
+        #[unstable(feature = "int_fetch_ops", issue = "none")]
+        pub fn fetch_add(&mut self, val: Self) -> Self {
+            let old = *self;
+            *self = self.wrapping_add(val);
+            old
+        }
+
+        /// Subtracts from the current value, returning the previous value.
+        ///
+        /// This operation wraps around on overflow.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// #![feature(int_fetch_ops)]
+        #[doc = concat!("let mut foo: ", stringify!($SelfT), " = 20;")]
+        /// assert_eq!(foo.fetch_sub(10), 20);
+        /// assert_eq!(foo, 10);
+        /// ```
+        #[inline]
+        #[unstable(feature = "int_fetch_ops", issue = "none")]
+        pub fn fetch_sub(&mut self, val: Self) -> Self {
+            let old = *self;
+            *self = self.wrapping_sub(val);
+            old
+        }
+
+        /// Bitwise "and" with the current value, returning the previous value.
```

Comment:
As a bonus, would it be possible to link to the relevant traits for the operations that have one (and, not, or, ...) ? I assume people looking for fetch_and would know what 'and' is but more links is always nice and make exploring the docs easier for new people

Reply:
Sure, I'll add this.